src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,67 @@
+import { posix } from 'path';
+import { types } from 'mime-types';
+import { RuntimeConfig } from '../init/RuntimeConfig';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { APPLICATION_OCTET_STREAM } from '../util/ContentTypes';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { trimTrailingSlashes } from '../util/Util';
+import { FileIdentifierMapper } from './FileIdentifierMapper';
+
+const { join: joinPath } = posix;
+
+export class ExtensionBasedMapper implements FileIdentifierMapper {
+  private readonly baseRequestURI: string;
+  private readonly rootFilepath: string;
+  private readonly types: Record<string, any>;
+
+  public constructor(runtimeConfig: RuntimeConfig, overrideTypes = { acl: 'text/turtle', metadata: 'text/turtle' }) {
+    this.baseRequestURI = trimTrailingSlashes(runtimeConfig.base);
+    this.rootFilepath = trimTrailingSlashes(runtimeConfig.rootFilepath);
```

Comment:
This is not going to work. The idea behind the `RuntimeConfig` is that its values get filled in at runtime, not at construction time (due to how the dependency injection works), so at this point they would still be empty. (Until #106 gets resolved that is).

Reply:
We could make them getters for now (let's make a comment note in the code though with ref to #106)

Reply:
ah indeed they were getters, so I will change it back